app/abilities/report.js

```diff
@@ -1,32 +1,34 @@
-import { computed } from "@ember/object";
+import { inject as service } from "@ember/service";
+import { tracked } from "@glimmer/tracking";
 import { Ability } from "ember-can";
 
-export default Ability.extend({
-  canEdit: computed(
-    // eslint-disable-next-line ember/use-brace-expansion
-    "model.user.{id,supervisors}",
-    "model.verifiedBy.id",
-    "model.{billed,customerAssignees,projectAssignees,taskAssignees}",
-    "user.{id,isSuperuser}",
-    function () {
-      const isEditable =
-        this.user?.isSuperuser ||
-        (!this.model?.verifiedBy?.get("id") &&
+export default class ReportAbility extends Ability {
+  @service session;
+  @tracked user;
+
+  constructor(...args) {
+    super(...args);
+    this.user = this.session.data.user;
+  }
```

Comment:
Why not using a getter for the `user` as in the other ability classes?

Reply:
because of tests in `tests/unit/abilities/report-test.js` file
this file is setting the value of `user` property

Reply:
I think the test should not be the priority for a the actual implementation. Let's try it with a `get user ()` in the ability and change the test to use `setupSessionMock`

```js
import setupSession from "timed/tests/helpers/session-mock";

// in the test module
setupSession(hooks)

// in the test case
const session = this.owner.lookup("service:session");
session.data.user = { ...whateverisneededhere }
```